electrode_models/nucleation_single_particle/initialize.py

```diff
@@ -0,0 +1,129 @@
+        params, offset):
+    """
+    Initialize the model.
+    """ 
+    class electrode:
+        """
+        Create an electrode object representing the dense electrode
+        """
+
+        # Import relevant Cantera objects.
+        bulk_obj = ct.Solution(input_file, inputs['bulk-phase'])
+        elyte_obj = ct.Solution(input_file, inputs['electrolyte-phase'])
+        conductor_obj = ct.Solution(input_file, inputs['conductor-phase'])
+        surf_obj = ct.Interface(input_file, inputs['surf-phase'], 
+            [bulk_obj, elyte_obj, conductor_obj])
+
+        # Anode or cathode? Positive external current delivers positive charge 
+        # to the anode, and removes positive charge from the cathode.
+        name = electrode_name
+        if name=='anode':
+            i_ext_flag = -1
+        elif name=='cathode':
+            i_ext_flag = 1
+        else:
+            raise ValueError("Electrode must be an anode or a cathode.")
+
+        # Store the species index of the Li ion in the Cantera object for the 
+        # electrolyte phase:
+        index_Li = elyte_obj.species_index(inputs['mobile-ion'])
+
+        # Electrode thickness and inverse thickness:
+        dy = inputs['thickness']
+        dyInv = 1/dy
+
+        # Phase volume fractions
+        eps_solid = inputs['eps_solid']
+        eps_elyte = 1 - eps_solid
+
+        # Electrode-electrolyte interface area, per unit geometric area.
+        # This calculation assumes spherical particles of a single radius, with 
+        # no overlap.
+        A_surf_ratio = (3*eps_solid*dy/inputs['r_p'])
+
+        # Inverse double layer capacitance, per unit interfacial area.
+        C_dl_Inv = 1/inputs['C_dl']
+
+        # Microstructure-based transport scaling factor, based on Bruggeman 
+        # coefficient of -0.5:
+        elyte_microstructure = eps_elyte**1.5
+        
+        # SV_offset specifies the index of the first SV variable for the 
+        # electrode (zero for anode, nVars_anode + nVars_sep for the cathode)
+        SV_offset = offset
+
+        # Determine the electrode capacity (Ah/m2)
+
+        # Max concentration of stored ion (intercalated Li)
+        # Save initial X
+        X_o = bulk_obj.X 
+        # Set object concentration to fully lithiated:
+        bulk_obj.X = inputs['stored-ion']['name']+':1.0' 
+        # Concentration of stored Li, per unit volume of intercalation phase:
+        C = bulk_obj[inputs['stored-ion']['name']].concentrations[0]
+        
+        capacity = (C*inputs['stored-ion']['charge']*ct.faraday
+                *inputs['eps_solid'])*inputs['thickness']/3600
+
+        # Return Cantera object composition to original value:
+        bulk_obj.X = X_o
+        
+        # Number of state variables: electrode potential, electrolyte 
+        # potential, electrode composition (n_species), electrolyte composition 
+        # (n_species)
+        nVars = 2 + bulk_obj.n_species + elyte_obj.n_species + inputs['buckets']
```

Comment:
You will want to overwrite this.  How many variables need to be stored?  if you are implementing your HNG code, it will be `2 + nsp_elyte + n_bins`, correct?

- 1 for electrode phase electric potential
- 1 for electrolyte phase electric potential (or double layer potential)
- 1 for each species in the electrolyte (molar concentration)
- 1 for the number of particles in each histogram bin

This also means the user will need to input info for the histogram (upper and lower radius limits, number of bins)